src/extension-runners/chromium.js

```diff
@@ -292,10 +298,30 @@ export class ChromiumExtensionRunner {
   async reloadExtensionBySourceDir(
     extensionSourceDir: string // eslint-disable-line no-unused-vars
   ): Promise<Array<ExtensionRunnerReloadResult>> {
-    // TODO(rpl): detect the extension ids assigned to the
-    // target extensions and map it to the extensions source dir
-    // (https://github.com/mozilla/web-ext/issues/1687).
-    return this.reloadAllExtensions();
+    const runnerName = this.getName();
+    const extensionId = this.generateId(extensionSourceDir);
```

Comment:
Extensin ID does not only depend on the directory location; the manifest is used too, and takes precedence.

For an example, search for `publicKeyToExtensionId` in https://github.com/Rob--W/crxviewer/blob/30ae3c4e7acd495f4bb87d2f74af4496aa237847/src/crxviewer.js